Packs/AzureActiveDirectory/Integrations/AzureADIdentityProtection/AzureADIdentityProtection.py

```diff
@@ -215,6 +251,96 @@ def azure_ad_identity_protection_risky_users_dismiss_command(client: AADClient,
     return client.azure_ad_identity_protection_risky_users_dismiss(**kwargs)
 
 
+def create_incidents_from_input(input: List[Dict[str, str]], last_fetch_datetime: datetime) -> \
+        Tuple[List[Dict[str, str]], datetime]:
+
+    incidents: List[Dict[str, str]] = []
+    last_fetch = last_fetch_datetime
+
+    for current_input in input:
+        # 'activityDateTime': '2021-07-15T11:02:54Z' / 'activityDateTime': '2021-07-15T11:02:54.12345Z'
+        activity_date_time_str: str = current_input.get('activityDateTime', '')
+
+        activity_datetime = dateparser.parse(activity_date_time_str)
+        # to prevent duplicates, adding incidents with creation_time > last fetched incident
+        if last_fetch:
+            if activity_datetime <= last_fetch:
+                demisto.debug(f'{INTEGRATION_NAME} - alert {str(current_input)} created at {activity_date_time_str}.'
+                              f' Skipping.')
+                continue
+
+        current_id: str = current_input.get('id', '')
+        current_risk_event_type: str = current_input.get('riskEventType', '')
+        current_risk_detail: str = current_input.get('riskDetail', '')
+        incident = {
+            'name': f'{INTEGRATION_NAME} Incident:'
+                    f' {current_risk_event_type} {current_risk_detail} {current_id}',
+            'occurred': activity_date_time_str,
+            'rawJSON': json.dumps(current_input)
+        }
+        incidents.append(incident)
+
+        activity_date_time_str = activity_date_time_str.split(".")[0]
+        if not activity_date_time_str.endswith("Z"):
+            # microseconds, 'activityDateTime': '2021-07-15T11:02:54.12345Z'
+            activity_date_time_str += "Z"
+
+        # timestamp = datetime.strptime(activity_date_time_str, DATE_FORMAT)
+        timestamp = activity_datetime
+        if timestamp > last_fetch:
+            last_fetch = timestamp
+
+    return incidents, last_fetch
+
+
+def fetch_incidents(client: AADClient, params: Dict[str, str]):
+
+    last_run: Dict[str, str] = demisto.getLastRun()
+    demisto.debug(f'last run: {last_run}')
+
+    last_fetch = last_run.get('last_item_time', '')
+    if not last_fetch:
+        # handle first time fetch
+        FETCH_TIME = params.get('first_fetch', '1 days')
+        default_fetch_datetime, _ = parse_date_range(date_range=FETCH_TIME, utc=True, to_timestamp=False)
```

Comment:
Why using upper case letters for `FETCH_TIME`?

Reply:
Renamed to `first_fetch`, the param's name. https://github.com/demisto/content/pull/14352/commits/260c3909b15d17e9cbd445d61ba8a725103f05ba